File: src/server/InvestmentApp-Server/Models/Experts/ExpertProject.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace InvestmentApp.Models.Experts;

public class ExpertProject
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; }

    public Expert Expert { get; set; }

    public Guid ExpertId { get; set; }

    public Project Project { get; set; }

    public Guid ProjectId { get; set; }

    public Period Period { get; set; }

    public Guid PeriodId { get; set; }

    public Possibility Possibility { get; set; }

    public Guid PossibilityId { get; set; }

    public double CashFlowRate { get; set; }
}

File: src/client/src/app/models/expert-project.ts
import { DataTableRow, Period, Possibility, Project } from ".";
import { Guid } from "guid-typescript";

// TODO add expert
export interface ExpertProject {
    id: Guid
    cashFlowRate: number;
    possibility: Possibility;
    possibilityId: Guid;
    period: Period;
    periodId: Guid;
    project: Project;
    projectId: Guid;
}

export interface ExpertProjectRow extends DataTableRow {
    id: string,
    internalId: string;
    cashFlowRate: number;
    possibility: string;
    possibilityId: string;
    period: string;
    periodId: string;
    project: string;
    projectId: string;
}
